### ISSUE TITLE
why 10000 in your code,what's the meaning?Thanks!!!

### ISSUE THREAD

**Issue by liuyishoua** [carlini/nn_robust_attacks#48]
in l2 completion,you use codes like below. What is the meaning?And why you use 10000?Could you tell me some more detail?

other = tf.reduce_max((1-self.tlab)*self.output - (self.tlab*10000),1)

**Comment by fotinidelig** [comment]
(For anyone still interested) 
I think it's just to make sure that this max values doesn't take into account the target label. 'Cause here you want to calculate the maximum logit of the predictions WITHOUT the target (a.k.a if the network is accurate enough, should correspond to the true label of the images).
Personally I'm not sure it makes any difference even if logit[tlab] = 1 and logit[label!=tlab] = 0.
Also note that tlab = [target label], real = [logit of target label] & other = [max logit between all other labels]

**Comment by liuyishoua** [comment]
this help a lot,thank you
